lib/src/utilities/extract-nested-types.ts

```diff
@@ -0,0 +1,127 @@
+import assertNever from "assert-never";
+import { flatten } from "lodash";
+import { TypeNode } from "../models/nodes";
+import { ObjectType, TypeKind, UnionType } from "../models/types";
+import { resolveType } from "../verifiers/utilities/type-resolver";
+
+export function extractNestedUnionTypes({
```

Comment:
It would be worth writing some JSDoc that explains that nested reference types are not included. This is a change of behaviour compared to the previous implementation.